src/jdk.hotspot.agent/share/native/libsaproc/sadis.c

```diff
@@ -156,8 +147,7 @@ JNIEXPORT jlong JNICALL Java_sun_jvm_hotspot_asm_Disassembler_load_1library(JNIE
 #else
   hsdis_handle = dlopen(libname, RTLD_LAZY | RTLD_GLOBAL);
   if (hsdis_handle == NULL) {
-    snprintf(buffer, sizeof(buffer), "%s%s", jrepath, libname);
-    hsdis_handle = dlopen(buffer, RTLD_LAZY | RTLD_GLOBAL);
+    hsdis_handle = dlopen(libname, RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
If the first dlopen fails, then you are just repeating it. Same for the windows code just above this. I think the second attempt is no longer needed since you are now dealing with a libname that you know exists due to the changes in Disassembler.java.